Configurations/unix-Makefile.tmpl

```diff
@@ -506,6 +506,7 @@ install_dev: install_runtime_libs
 	@chmod 644 $(DESTDIR)$(INSTALLTOP)/include/openssl/applink.c
 	@ : {- output_on() unless grep { $_ eq "OPENSSL_USE_APPLINK" } (@{$target{defines}}, @{$config{defines}}); "" -}
 	@set -e; for i in $(SRCDIR)/include/openssl/*.h \
+			  $(SRCDIR)/crypto/include/internal/*.h \
```

Comment:
Absolutely not. Internal headers should *not* get publically installed, that's asking for trouble.

Reply:
It's a mistake. What i really want to copy is /include/openssl/sm*.h instead  of those internal header file.  Because i found when after "make install" finished, in the install directory, there is only  sm2_sign( ) in libcrypto.a , but no definition header file in "include/openssl". If i want to invoke the function of sm2_sign, i don't have a header file to included. So i just did it  as RSA_sign define in "include/openssl/rsa.h" did,  copy the rsa.h to "include/openssl" directory".
Correct me if i misunderstood the usage!

Reply:
You do misunderstand the usage. We have decided that for any new algorithm, the EVP API will be the only way to access it. That is even more important with the upcoming 3.0 release. That's the reason you will not find any public header file for the sm algos.